providers/implementations/keymgmt/dsa_kmgmt.c

```diff
@@ -430,6 +430,7 @@ static void *dsa_gen_init(void *provctx, int selection,
         OSSL_FIPS_IND_INIT(gctx)
     }
     if (!dsa_gen_set_params(gctx, params)) {
+        OPENSSL_free(gctx->mdname);
```

Comment:
same comment applies here also... Call the cleanup function instead.

Reply:
Changed here too, thanks